includes/Preferences.php

```diff
@@ -109,7 +109,15 @@ static function getPreferences( $user, IContextSource $context ) {
 	 * @return array|String
 	 */
 	static function getUserPreference( $name, $info, $user ) {
-		$val = $user->getGlobalPreference( $name );
+		$getCallback = function($property) use ($user) {
+			if (in_array($property, self::getAttributes())) {
+				return $user->getGlobalAttribute($property);
+			} else {
+				return $user->getGlobalPreference($property);
+			}
+		};
```

Comment:
For clarity, It seems like this should be local function and not an anonymous one.